service/controller/clusterapi/v17/resources/clusterid/resource.go

```diff
@@ -5,28 +5,35 @@ import (
 	"github.com/giantswarm/microerror"
 	"github.com/giantswarm/micrologger"
 	"sigs.k8s.io/cluster-api/pkg/client/clientset_generated/clientset"
+
+	"github.com/giantswarm/cluster-operator/service/controller/clusterapi/v17/key"
 )
 
 const (
 	Name = "clusteridv17"
 )
 
 type Config struct {
-	CMAClient clientset.Interface
-	G8sClient versioned.Interface
-	Logger    micrologger.Logger
+	CMAClient                   clientset.Interface
+	CommonClusterStatusAccessor key.CommonClusterStatusAccessor
```

Comment:
I think this thing does not need to be configured at all.

Reply:
Well, not necessarily right now, but it does when there is another provider. I don't think this hurts here.